src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -561,8 +565,35 @@ scaleNominalToMergeCredit (NominalDebt             nominalDebt)
     let mergeCredits_spec = floor $ toRational nominalCredits
                                   * toRational mergeDebt
                                   / toRational nominalDebt
+        mergeCredits_fast = w2i $ timesDivABC_fast (i2w nominalCredits)
+                                                   (i2w mergeDebt)
+                                                   (i2w nominalDebt)
```

Comment:
We could use some text from the commit message to explain the reasoning behind using the fast division. 

Also, the commit message mentions something about 32 bits, but we assume a machine word is 64 bits, so is this a typo?

Reply:
done